components/esp32/freertos_hooks.c

```diff
@@ -16,22 +16,24 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <stdbool.h>
+#include "freertos/FreeRTOS.h"
 #include "esp_attr.h"
 #include "esp_freertos_hooks.h"
 
 //We use just a static array here because it's not expected many components will need
 //an idle or tick hook.
 #define MAX_HOOKS 8
 
-static esp_freertos_idle_cb_t idle_cb[MAX_HOOKS]={0};
-static esp_freertos_tick_cb_t tick_cb[MAX_HOOKS]={0};
+static esp_freertos_idle_cb_t idle_cb[2][MAX_HOOKS]={0};
+static esp_freertos_tick_cb_t tick_cb[2][MAX_HOOKS]={0};
```

Comment:
Could you use portNUM_PROCESSORS instead of '2' here?

Reply:
OK, will fix.